MatchServer/MMatchRuleGunGame.cpp

```diff
@@ -0,0 +1,134 @@
+#include "stdafx.h"
+#include "MMatchRuleGunGame.h"
+#include "MBlobArray.h"
+
+MMatchRuleGunGame::MMatchRuleGunGame(MMatchStage* pStage) : MMatchRule(pStage)
+{
+}
+
+MMatchRuleGunGame::~MMatchRuleGunGame()
+{
+}
+
+void MMatchRuleGunGame::OnBegin()
+{
+	if (m_pGGSet.size() > 0) {
+		for (auto itor = m_pGGSet.begin(); itor != m_pGGSet.end(); ++itor)
+		{
+			delete *itor;
+		}
+		m_pGGSet.clear();
+	}
+
+	int randSet = rand() % MGetGunGame()->GetSetCount();
```

Comment:
Yeah, I understand that. I think I expressed myself wrong. What I meant was that since you're only using the IDs in this way, you could just remove the ID tags from the XML file completely and just load each SET into an element of `std::vector<std::vector<GGSet>> m_GGSet.` Then, `m_GGSet[rand() % m_GGSet.size()]` would still give you a random `std::vector<GGSet>`, without having a superfluous map and some IDs that seem to be pointless. Does that make sense?

Also, the way you use "GGSet" is pretty confusing. First, there's a struct called "GGSet," then there's a `std::vector<GGSet*>` called "m_pGGSet," and then there's a `std::map<int, std::vector<GGSet*>` *also* called "m_GGSet." :p